source/Dialog.cpp

```diff
@@ -224,7 +224,22 @@ bool Dialog::KeyDown(SDL_Keycode key, Uint16 mod, const Command &command, bool i
 {
 	auto it = KEY_MAP.find(key);
 	bool isCloseRequest = key == SDLK_ESCAPE || (key == 'w' && (mod & (KMOD_CTRL | KMOD_GUI)));
-	if((it != KEY_MAP.end() || (key >= ' ' && key <= '~')) && !isMission && (intFun || stringFun) && !isCloseRequest)
+	if(stringFun && (mod & KMOD_CTRL) && (key == 'c' || key == 'v'))
+	{
+		if(key == 'c')
+			SDL_SetClipboardText(input.c_str());
+		else if(SDL_HasClipboardText())
+		{
+			input.clear();
```

Comment:
I understand from the description that you wanted whole-field paste, but I don't see why that's necessary. The only necessary change to allow for additive paste would be to remove this line, and make the char limit code account for the total input length (if we want to keep that at all).

Reply:
In most modern programs pasting doesn't overwrite the previous contents, so I think it would be actually _more_ intuitive.